### PR TITLE
[ci] Avoid running ROM tests that have a corresponding sival test

### PR DESCRIPTION
**Pull request by pamaury**
This PR reduces the number of ROM tests and favors the Sival environment for tests that exists both in ROM and SiVal.

### CODE REVIEW

**Comment by pamaury** [comment]
The CI failure is unrelated. This PR reduces the number of tests quite a bit: 297 instead of 375 previously. The time gain does not seem to be massive (maybe 5 to 10 minutes) but I think that's worth it.

File: ci/scripts/run-fpga-tests.sh [hunk 1]
```diff
@@ -50,41 +50,66 @@ test_args=""
...
+            kind(
+                \"test_suite\",
+                //...
+            )
+            except
+            `# Remove all test suites depending on a test tagged cw310_sival_rom_ext`
+            rdeps(
+                //...,
+                `# Find all tests tagged cw310_sival_rom_ext`
+                attr(\"tags\",\"cw310_sival_rom_ext\", //...),
```
Comment (nbdd0121) [inline ci/scripts/run-fpga-tests.sh:75]: should this check for broken?

Comment (pamaury) [reply ci/scripts/run-fpga-tests.sh:75]: Hum, that's a good point, we might not want to ignore a test if the corresponding `cw310_sival_rom_ext` is marked as broken? I will update this.

**Review by moidx — APPROVED** [review]

**Comment by pamaury** [comment]
Thanks @moidx for the comment, indeed I think that we should filter out `manual` tests because previously we were using a wildcard pattern but now we are listing targets explicitely. I will change that.

**Comment by pamaury** [comment]
Can we merge this?

**Comment by pamaury** [comment]
> Reference in new iss

I agree, although I think this will be hard to achieve with our current approach. I was thinking that maybe a better approach would be to have a "test dispatcher": a small program (python) that queries all the tests and assign them to the various test job in the CI. This would make it much easier to make sure that every test is dispatched and that there is minimal redundancy between execution environments.

**Pull request merged**